### PR TITLE
Add Pushupmaster

### PR DESCRIPTION
**Pull request by tronta**
### Please confirm your submission meets all the criteria

- [xI have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.gnome.pushupmaster.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+    "app-id" : "org.gnome.pushupmaster",
```
Comment (hfiguiere) [inline org.gnome.pushupmaster.json:2]: It this an official gnome app?

File: org.gnome.pushupmaster.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "master",
```
Comment (hfiguiere) [inline org.gnome.pushupmaster.json:4]: this version is not available on flathub

File: org.gnome.pushupmaster.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "pushupmaster",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--socket=pulseaudio",
+        "--filesystem=/sys/bus/iio/devices/"
```
Comment (hfiguiere) [inline org.gnome.pushupmaster.json:16]: this is not valid.

File: org.gnome.pushupmaster.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
```
Comment (hfiguiere) [inline org.gnome.pushupmaster.json:21]: this won't work. You need to use flatpak-builder-tools to generate a manifest for the dependencies. Or have all the crates vendored in the tarball.

https://github.com/flatpak/flatpak-builder-tools

File: org.gnome.pushupmaster.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    "modules" : [
+        {
+            "name" : "pushupmaster",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/peter/pushupmaster.git",
+                    "branch": "main"                }
```
Comment (hfiguiere) [inline org.gnome.pushupmaster.json:48]: you need a revision or tag. Not a branch.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.gnome.pushupmaster.json (hunk 2 above)
Comment (tronta) [reply org.gnome.pushupmaster.json:4]: ah. ok. I need to check then, if it works with an older version, or I have to wait,

File: org.gnome.pushupmaster.json (hunk 1 above)
Comment (tronta) [reply org.gnome.pushupmaster.json:2]: actually no, but I put it is on the git.gnome repo.
If I should change sth there, please tell me. It's my first app :-)

File: org.gnome.pushupmaster.json (hunk 3 above)
Comment (hfiguiere) [reply org.gnome.pushupmaster.json:16]: it's not about this change. it's about `/sys`

Comment (tronta) [reply org.gnome.pushupmaster.json:16]: I need this to access the proximity sensor on the phone. What would be the correct way?

Comment (nedrichards) [reply org.gnome.pushupmaster.json:16]: Does your device expose information via https://gitlab.freedesktop.org/hadess/iio-sensor-proxy/ ?

File: org.gnome.pushupmaster.json (hunk 1 above)
Comment (nedrichards) [reply org.gnome.pushupmaster.json:2]: Yup, this isn't GNOME software as released by the GNOME release team. Check out https://circle.gnome.org/ for ways to get your app a 'seal of approval'.

Comment (tronta) [reply org.gnome.pushupmaster.json:2]: Thanks for the hint. I wanted to do this anyway. Let me work on the other topics and some additional improvements on the SW and I will go for it.

**Pull request closed without merging**